### PR TITLE
fix: Use types instead of interfaces

### PR DESCRIPTION
**Pull request by jjtang1985**
## Context

Due to some eslint rules, we used interfaces for some of the orchestration types by accident.
In order to keep the consistent UX, we will keep the original generated client.

- [x] wait for #270 first.

## Definition of Done

- [x] Code is tested (Unit, E2E)
- [x] (Optional) Release notes updated

### CODE REVIEW

File: .changeset/interface-vs-type.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+'@sap-ai-sdk/orchestration': minor
+---
+[Compatibility Note] Switch some of the orchestration interfaces to types, which was introduced by accident.
```
Comment (deekshas8) [inline .changeset/interface-vs-type.md:4]:
```suggestion
[Compatibility Note] Switch some of the orchestration interfaces to types, as they were introduced by accident.
```

File: packages/orchestration/src/client/api/schema/module-results.ts [hunk 2]
```diff
@@ -10,12 +10,12 @@ import type { LlmChoice } from './llm-choice.js';
 /**
  * Results of each module.
  */
-export interface ModuleResults {
+export type ModuleResults = {
   grounding?: GenericModuleResult;
```
[the remark this answers is not in the document]
Comment (deekshas8) [reply packages/orchestration/src/client/api/schema/module-results.ts:14]: This would also need a compatibility note.

**Review by deekshas8 — APPROVED** [review]
Already approving this (have disabled auto-merge). Please have a look into grounding in output.

File: packages/orchestration/src/client/api/schema/module-results.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jjtang1985) [reply packages/orchestration/src/client/api/schema/module-results.ts]: - removed the grounding key
- added compatibility note

**Pull request merged**